components/eam/src/physics/cam/convect_deep.F90

```diff
@@ -1,428 +1,331 @@
-
-! Arguments
-   type(physics_state), intent(in ) :: state          ! Physics state variables
-   type(physics_ptend), intent(out) :: ptend          ! indivdual parameterization tendencies
-   
-   type(physics_buffer_desc), pointer :: pbuf(:)
-
-   real(r8), intent(in) :: ztodt                          ! 2 delta t (model time increment)
-   real(r8), intent(in):: mu(pcols,pver) 
-   real(r8), intent(in):: eu(pcols,pver) 
-   real(r8), intent(in):: du(pcols,pver) 
-   real(r8), intent(in):: md(pcols,pver) 
-   real(r8), intent(in):: ed(pcols,pver) 
-   real(r8), intent(in):: dp(pcols,pver) 
-   
-   ! wg layer thickness in mbs (between upper/lower interface).
-   real(r8), intent(in):: dsubcld(pcols) 
-   
-   ! wg layer thickness in mbs between lcl and maxi.    
-   integer, intent(in) :: jt(pcols)   
-   
-   ! wg top  level index of deep cumulus convection.
-   integer, intent(in) :: maxg(pcols) 
-   
-   ! wg gathered values of maxi.
-   integer, intent(in) :: ideep(pcols)
-   
-   ! w holds position of gathered points vs longitude index   
-   integer, intent(in) :: lengath
-
-   integer, intent(in) :: species_class(:)
-
-   if ( deep_scheme .eq. 'ZM' ) then  !    1 ==> Zhang-McFarlane (default)
-      call zm_conv_tend_2( state,   ptend,  ztodt,  pbuf,mu, eu, &
-     du, md, ed, dp, dsubcld, jt, maxg, ideep, lengath, species_class) 
-   else
+   use constituents,    only: pcnst
+   use zm_conv_intr,    only: zm_conv_tend_2
+   !----------------------------------------------------------------------------
+   ! Arguments
+   type(physics_state),             intent(in ) :: state          ! physics state
+   type(physics_ptend),             intent(out) :: ptend          ! output tendencies
+   real(r8),                        intent(in ) :: ztodt          ! 2 delta t (model time increment)
+   type(physics_buffer_desc),       pointer     :: pbuf(:)        ! physics buffer
+   real(r8), dimension(pcols,pver), intent(in ) :: mu             ! upward cloud mass flux               [?]
+   real(r8), dimension(pcols,pver), intent(in ) :: eu             ! entrainment in updraft               [?]
+   real(r8), dimension(pcols,pver), intent(in ) :: du             ! detrainment in updraft               [?]
+   real(r8), dimension(pcols,pver), intent(in ) :: md             ! downward cloud mass flux             [?]
+   real(r8), dimension(pcols,pver), intent(in ) :: ed             ! entrainment in downdraft             [?]
+   real(r8), dimension(pcols,pver), intent(in ) :: dp             ! layer thickness between interfaces   [mb]
+   real(r8), dimension(pcols),      intent(in ) :: dsubcld        ! layer thickness between lcl and maxi [mb]
+   integer,  dimension(pcols),      intent(in ) :: jt             ! top level index of deep convection
+   integer,  dimension(pcols),      intent(in ) :: maxg           ! gathered values of maxi
+   integer,  dimension(pcols),      intent(in ) :: ideep          ! flag to indicate active column for gathering
```

Comment:
descriptions in comments don't match up here either for `dsubcld`, `jt`, `maxg`, and `ideep`